plugins/zendesk/icon_zendesk/connection/connection.py

```diff
@@ -21,7 +22,9 @@ def connect(self, params):
         elif params.get("api_key").get("secretKey"):
             creds["token"] = params.get("api_key").get("secretKey")
         else:
-            raise Exception("Error: Must define either password or API key!")
+            raise PluginException(
+                cause="Could not authenticate to Zendesk.", assistance="Please provide a password or API key."
+            )
```

Comment:
This class should have a `test` function for connection tests

Reply:
Connection test method added. Locally tested:

```python
>>> import zenpy
>>> creds = {"email": "GOOD_CREDENTIALS"}
>>> client = zenpy.Zenpy(**creds)
>>> client.users()
<zenpy.lib.generator.ZendeskResultGenerator object at 0x1038e7090>
>>> creds = {"email": "BAD_CREDENTIALS"}
>>> client = zenpy.Zenpy(**creds)
>>> client.users()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mberezin/go/src/github.com/rapid7/insightconnect-plugins/plugins/zendesk/venv/lib/python3.7/site-packages/zenpy/lib/api.py", line 350, in __call__
    **kwargs)
  File "/Users/mberezin/go/src/github.com/rapid7/insightconnect-plugins/plugins/zendesk/venv/lib/python3.7/site-packages/zenpy/lib/api.py", line 283, in _query_zendesk
    endpoint=endpoint(*endpoint_args, **endpoint_kwargs)))
  File "/Users/mberezin/go/src/github.com/rapid7/insightconnect-plugins/plugins/zendesk/venv/lib/python3.7/site-packages/zenpy/lib/api.py", line 108, in _get
    **kwargs)
  File "/Users/mberezin/go/src/github.com/rapid7/insightconnect-plugins/plugins/zendesk/venv/lib/python3.7/site-packages/zenpy/lib/api.py", line 150, in _call_api
    self._check_response(response)
  File "/Users/mberezin/go/src/github.com/rapid7/insightconnect-plugins/plugins/zendesk/venv/lib/python3.7/site-packages/zenpy/lib/api.py", line 307, in _check_response
    raise APIException(json.dumps(_json), response=response)
zenpy.lib.exception.APIException: {"error": "Couldn't authenticate you"}
```